### PR TITLE
chore: start the image builds sooner to speed up PR build times

### PR DESCRIPTION
**Pull request by chirino**
try to implement a better wait condition for the crunchy postgres to be up
try to enable cockroach based e2e

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -233,21 +232,21 @@ jobs:
...
         # todo: work on getting the cockroach overlay working ["dev", "cockroach"]
-        overlay: ["dev"]
+        overlay: ["dev", "cockroach"]
 
     steps:
       - uses: actions/checkout@v3
       - uses: docker/setup-buildx-action@v2
 
-      - name: Setup Go
+      - name: Setup Gok
```
Comment (russellb) [inline .github/workflows/build.yml:249]: typo?

File: Makefile [hunk 2]
```diff
@@ -335,13 +335,13 @@ redeploy: images load-images ## Redeploy nexodus after images changes
 init-db:
 # wait for the DB to be up, then restart the services that use it.
 ifeq ($(OVERLAY),dev)
-# todo: find a reliable way to wait for the crunchy based deployment of postgres to be online
+	$(CMD_PREFIX) kubectl wait -n nexodus crdbcluster/cockroachdb --timeout=15m --for=condition=StanzaCreated || true
```
Comment (russellb) [inline Makefile:338]: you've got cockroachdb here in the normal dev overlay (postgres)

Comment (chirino) [reply Makefile:338]: doh

File: .github/workflows/build.yml (hunk 1 above)
Comment (chirino) [reply .github/workflows/build.yml:249]: yep

**Pull request merged**